optimum/habana/trl/trainer/sft_trainer.py

```diff
@@ -385,3 +396,140 @@ def make_inputs_require_grad(module, input, output):
                 # Remove the PrinterCallback to avoid duplicated prints in case we passed a `RichProgressCallback`
                 if callback.__class__.__name__ == "PrinterCallback":
                     self.callback_handler.pop_callback(callback)
+
```

Comment:
Could you import the function from SFTTrainer and made the change for pad_max or maybe make the change in SFTTrainer class itself instead of duplicating the code here?

Reply:
@lkk12014402 , pls help work on it, thx.

Reply:
@lkk12014402 , pls mark it as resolved if you resolved this issue, thx.

Reply:
ok, I am working on it

Reply:
is it done?

Reply:
hi @emascarenhas `SFTTrainer` is imported from `trl`. and `pad_max` works on gaudi, so I rewrite the function based on `SFTTrainer`.